rpc/http.go

```diff
@@ -36,11 +36,14 @@ import (
 )
 
 const (
-	contentType             = "application/json"
 	maxRequestContentLength = 1024 * 512
 )
 
-var nullAddr, _ = net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+var (
+	acceptedContentTypes = []string{"application/json", "application/json-rpc"}
```

Comment:
```suggestion
	acceptedContentTypes = []string{"application/json", "application/json-rpc", "application/jsonrequest"}
```

let's directly add application/jsonrequest so we cover the full set given here: https://www.jsonrpc.org/historical/json-rpc-over-http.html#http-header